modules/pubwiseAnalyticsAdapter.js

```diff
@@ -10,44 +11,117 @@ const utils = require('src/utils');
  */
 
 const analyticsType = 'endpoint';
-let target_site = 'unknown';
-let target_url = 'https://staging.api.pubwise.io';
-let pw_version = '2.1.3';
+const analyticsName = 'PubWise Analytics: ';
+let defaultUrl = 'https://api.pubwise.io/api/v4/event/default/';
+let pubwiseVersion = '2.2';
+let configOptions = {site: '', endpoint: 'https://api.pubwise.io/api/v4/event/default/', debug: ''};
+let pwAnalyticsEnabled = false;
+let utmKeys = {'utm_source': '', 'utm_medium': '', 'utm_campaign': '', 'utm_term': '', 'utm_content': ''};
+
+let getParameterByName = function (name) {
+  var regexS = '[\\?&]' + name + '=([^&#]*)';
+  var regex = new RegExp(regexS);
+  var results = regex.exec(window.location.search);
+  if (results === null || results.length === 0) {
+    return '';
+  }
+
+  return decodeURIComponent(results[1].replace(/\+/g, ' '));
+};
+
+function markEnabled() {
+  utils.logInfo(analyticsName + 'Enabled', configOptions);
+  pwAnalyticsEnabled = true;
+}
+
+function enrichWithMetrics(dataBag) {
+  try {
+    dataBag['pw_version'] = pubwiseVersion;
+    dataBag['pbjs_version'] = pbjs.version;
+    dataBag['debug'] = configOptions.debug;
+  } catch (e) {
+    dataBag['error_metric'] = 1;
+  }
+
+  return dataBag;
+}
+
+function enrichWithUTM(dataBag) {
+  let newUtm = false;
+  try {
+    for (var prop in utmKeys) {
+      var urlValue = getParameterByName(prop);
+      utmKeys[prop] = urlValue;
+      if (utmKeys[prop] != '') {
+        newUtm = true;
+        dataBag[prop] = utmKeys[prop];
+      }
+    }
+
+    if (newUtm === false) {
+      for (var prop in utmKeys) {
+        var itemValue = localStorage.getItem('pw-' + prop);
+        if (itemValue.length !== 0) {
+          dataBag[prop] = itemValue;
+        }
+      }
+    } else {
+      for (var prop in utmKeys) {
+        localStorage.setItem('pw-' + prop, utmKeys[prop]);
+      }
+    }
+  } catch (e) {
+    dataBag['error_utm'] = 1;
+  }
+  return dataBag;
+}
+
+function sendEvent(eventType, data) {
+  utils.logInfo(analyticsName + 'Event ' + eventType + ' ' + pwAnalyticsEnabled, data);
```

Comment:
Hey, I think you misunderstood me. I meant use template literals. 
Check here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

e.g: ``utils.logWarn(`Test string ${variable}.`);``